ste/mgr-JobPartTransferMgr.go

```diff
@@ -97,6 +97,14 @@ type IJobPartTransferMgr interface {
 	ResetSourceSize() // sets source size to 0 (made to be used by setProperties command to make number of bytes transferred = 0)
 	SuccessfulBytesTransferred() int64
 	TransferIndex() (partNum, transferIndex uint32)
+	CheckPoint() ICheckpoint
+}
+
+type ICheckpoint interface {
+	Init(numChunks int)
+	ChunkDone(chunkID int)
+	TransferDone()
+	CompletedChunks() map[int]int
```

Comment:
To match the uncommitted list received from server we will take only completed chunks? we can take entire chunk list and save this additional map copy here. Just like 'ChunkDone' we can have a method 'IsChunkDone' where giving an index it just tells whether that chunk was uploaded from our end or not.